in_toto/runlib.py

```diff
@@ -570,6 +580,14 @@ def in_toto_run(name, material_list, product_list, link_cmd_args,
   if signature:
     signing_keyid = signature["keyid"]
     filename = FILENAME_FORMAT.format(step_name=name, keyid=signing_keyid)
+    
+    if metadata_directory is not None:
+      if not os.path.isdir(metadata_directory):
+        raise ValueError("--metadata-directory value is not a directory")
+      if not os.access(metadata_directory, os.W_OK):
+        raise PermissionError("Cannot write to {}".format(metadata_directory))
```

Comment:
I suggest to take an [EAFP](https://docs.python.org/3/glossary.html#term-eafp)-approach here, it's less code and more reliable. You can still enhance the error message by catching and re-raising the exception with a better message. If you do so I suggest to do it once in the dump function.

Reply:
I notice that both _in_toto_run.py_ and _in_toto_record.py_ have "try" and "except", which can catch the directory value error and permission error. So I decided to simply remove these two checks without changing the dump function.